File: Forma-MediaPipe/modules/expo-pose-detection/android/src/main/java/expo/modules/posedetection/ExpoPoseDetectionModule.kt
package expo.modules.posedetection

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoPoseDetectionModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoPoseDetection")

    Function("switchCamera") {
      ExpoPoseDetectionView.activeInstance?.switchCamera()
    }

    View(ExpoPoseDetectionView::class) {
      Events("onLandmark", "onVisionFrame")

      Prop("frameLimit") { view: ExpoPoseDetectionView, limit: Int ->
        view.configureFrameLimit(limit)
      }

      Prop("showSkeleton") { view: ExpoPoseDetectionView, show: Boolean ->
        view.configureShowSkeleton(show)
      }

      Prop("modelName") { view: ExpoPoseDetectionView, name: String ->
        view.configureModelName(name)
      }

      Prop("enableVisionDualEmit") { view: ExpoPoseDetectionView, enabled: Boolean ->
        view.configureVisionDualEmit(enabled)
      }
    }
  }
}

File: Forma-MediaPipe/modules/expo-pose-detection/ios/ExpoPoseDetectionModule.swift
import ExpoModulesCore

public class ExpoPoseDetectionModule: Module {
  static weak var activeView: ExpoPoseDetectionView?

  public func definition() -> ModuleDefinition {
    Name("ExpoPoseDetection")

    Function("switchCamera") {
      DispatchQueue.main.async {
        ExpoPoseDetectionModule.activeView?.switchCamera()
      }
    }

    View(ExpoPoseDetectionView.self) {
      Events("onLandmark", "onVisionFrame")

      Prop("frameLimit") { (view: ExpoPoseDetectionView, limit: Int) in
        view.configureFrameLimit(limit)
      }

      Prop("showSkeleton") { (view: ExpoPoseDetectionView, show: Bool) in
        view.configureShowSkeleton(show)
      }

      Prop("modelName") { (view: ExpoPoseDetectionView, name: String) in
        view.configureModelName(name)
      }

      Prop("enableVisionDualEmit") { (view: ExpoPoseDetectionView, enabled: Bool) in
        view.configureVisionDualEmit(enabled)
      }
    }
  }
}
